packages/flutter_tools/test/devices_test.dart

```diff
@@ -22,13 +23,15 @@ void main() {
       await createTestCommandRunner(command).run(<String>['devices']);
     });
 
-    testUsingContext('no error when no connected devices', () async {
-      final DevicesCommand command = new DevicesCommand();
-      await createTestCommandRunner(command).run(<String>['devices']);
-      expect(testLogger.statusText, contains('No devices detected'));
-    }, overrides: <Type, Generator>{
-      AndroidSdk: () => null,
-      DeviceManager: () => new DeviceManager(),
-    });
+    if (isRunningOnBot) {
+      testUsingContext('no error when no connected devices', () async {
+        final DevicesCommand command = new DevicesCommand();
+        await createTestCommandRunner(command).run(<String>['devices']);
+        expect(testLogger.statusText, contains('No devices detected'));
+      }, overrides: <Type, Generator>{
+        AndroidSdk: () => null,
+        DeviceManager: () => new DeviceManager(),
+      });
+    }
```

Comment:
I'll let @tvolkert and you figure out what should happen here, but if you do decide to skip it on bots, please use "skip:" so that it gets picked up by the technical debt benchmark. :-)